exer1-intf-table.py

```diff
@@ -1,17 +1,18 @@
 import json
 
-jsondata = open('exer1-interface-data.json').read()
+with open('exer1-interface-data.json', 'r') as jsondata:
+    interface_dict = json.load(jsondata)
 
-json_object = json.loads(jsondata)
-print(
-    "=======================================================================================" "\n"
-    "DN                                                 Description           Speed    MTU" "\n" 
-    "-------------------------------------------------- --------------------  ------  ------")
-imdata = json_object["imdata"]
-for i in imdata:
-        dn = i["l1PhysIf"]["attributes"]["dn"]
-        descr = i["l1PhysIf"]["attributes"]["descr"]
-        speed = i["l1PhysIf"]["attributes"]["speed"]
-        mtu = i["l1PhysIf"]["attributes"]["mtu"]
-        # print fields formatted in columns
-        print("{0:50} {1:20} {2:7} {3:6}".format(dn,descr,speed,mtu))
+imdata = interface_dict["imdata"]
+phys, attr = "l1PhysIf", "attributes"
+
+# List Comprehensions
```

Comment:
try to use list comprehensions whenever possible (don't sacrifice readability though)

https://stackoverflow.com/questions/30245397/why-is-list-comprehension-so-faster